autonomi/src/client/mod.rs

```diff
@@ -143,6 +140,10 @@ pub enum PutError {
     },
     #[error("Batch upload: {0}")]
     Batch(ChunkBatchUploadState),
+    #[error("Error occurred during network operation: {0}")]
+    NetworkError(#[from] NetworkError),
+    #[error("Error occurred during serialize/deserialize: {0}")]
+    ProtocolError(#[from] ant_protocol::Error),
```

Comment:
There is already a `NetworkError` wrapper above, as well as a `Serialization` error.
Those should be removed unless really necessary as they make error management upstream unpractical.